### PR TITLE
Created edges new, init, and free functions, and associated tests

### LINKED ISSUE #1446: Create new, init, and free functions for edges and tests for these functions
We have decided to make the edges field within the specgraph struct a dynamically allocated 2D array. However, this make it more difficult for the user to create the adjacency matrix. In this issue, we aim to create new, init, and free functions for the edges adjacency matrix within the specgraph struct to make it easier to create the adjacency matrix. We also aim to create tests for each of these functions.

### PR DESCRIPTION
**Pull request by williamhu314**
In this pull request, we have created new functions that provide functionality for creating a new pointer to an adjacency matrix (i.e. create the edges field in a specgraph, which is a heap-allocated 2D array), initializing the matrix, and freeing the matrix. We also made tests for each of these functions to ensure that they are working correctly.

The edges_new function simplifies the creation of an adjacency matrix for the game author. Rather than having them make a for loop to malloc each row of the matrix, they can just make a heap-allocated 1D array and then pass that into our edges_new function to get a heap-allocated 2D array.

The edges_free function simplifies freeing a heap-allocated 2D array as one would no longer need to use a for loop to free each individual row of the 2D array.

### CODE REVIEW

**Review by helenastc — APPROVED** [review]
This looks good, good job adding tests too. I would suggest maybe making a new issue to clear up the differences between edges and specgraph structs.

File: include/openworld/gen_structs.h [hunk 1]
```diff
@@ -294,6 +294,12 @@ roomspec_t* roomspec_new(char *room_name, char *short_desc, char *long_desc, ite
 */
 int roomspec_free(roomspec_t *spec);
 
+int edges_init(int** edges, int* inp_array, int num_rows, int num_cols);
```
Comment (philipchenaux) [inline include/openworld/gen_structs.h:310]: You must include proper function comments within the .h file. Please refer to other functions for a template to follow.

File: src/openworld/src/gen_structs.c [hunk 2]
```diff
@@ -191,6 +191,56 @@ roomspec_t* roomspec_new(char *room_name, char *short_desc, char *long_desc, ite
     return roomspecnew;
 }
 
+int edges_init(int** edges, int* inp_array, int num_rows, int num_cols)
```
Comment (philipchenaux) [inline src/openworld/src/gen_structs.c:195]: Again refer to other functions to see that you need to include a function comment. Specifically you should write /* see gen_structs.h */

**Review by philipchenaux — CHANGES_REQUESTED** [review]
A few minor syntax issues, but besides that I am happy with how the functions and tests look!

File: include/openworld/gen_structs.h (hunk 1 above)
Comment (rohandeme) [reply include/openworld/gen_structs.h:310]: Thanks for the reminder! Just added the comments.

File: src/openworld/src/gen_structs.c (hunk 2 above)
Comment (rohandeme) [reply src/openworld/src/gen_structs.c:195]: Just added those to the functions to the c files.

**Review by philipchenaux — APPROVED** [review]
Great job, looks good to merge!

**Comment by philipchenaux** [comment]
Issue Score: Excellent

Comments:
Excellent job on this PR!

**Pull request merged**